src/OpenSage.Game/Mathematics/Ray.cs

```diff
@@ -122,6 +98,11 @@ public Ray(Vector3 position, Vector3 direction)
             return result;
         }
 
+        public Ray Transform(Matrix4x4 world)
```

Comment:
This should have an `in` modifier to avoid copying the (large) `Matrix4x4` struct.